django_mongodb/compiler.py

```diff
@@ -57,11 +57,16 @@ def has_results(self):
 
     def get_converters(self, columns):
         converters = {}
-        for column in columns:
+        for name_column in columns:
+            try:
+                name, column = name_column
+            except TypeError:
+                # e.g., Count("*")
+                continue
             backend_converters = self.connection.ops.get_db_converters(column)
             field_converters = column.field.get_db_converters(self.connection)
             if backend_converters or field_converters:
-                converters[column.target.column] = backend_converters + field_converters
+                converters[name] = backend_converters + field_converters
```

Comment:
Django sql compiler uses an index. Like an integer and maintain the order.
https://github.com/mongodb-forks/django/blob/mongodb-5.0.x/django/db/models/sql/compiler.py#L1483

Reply:
Right, however, MongoDB returns a document of (unordered?) fields, hence we may need to use the key lookup approach (or at least this was the reason for the original implementation in Django Mongo Engine + Python 2). Since Python 3.7 dictionaries are guaranteed to be insertion ordered. Not sure how the MongoDB server handles field ordering though (does it match the order of `$projection`)? Could be checked but the idea is orthogonal to this PR which just builds on the existing implementation.